calico/network-policy/policy-rules/log-rules.mdx

```diff
@@ -0,0 +1,184 @@
+
+
+To make the most of log actions, follow these guidelines:  
+
+    1. **Order Matters:** Assign log actions a high `order` value to ensure they are evaluated after all higher-priority rules.  
+    1. **Explicit Allow Rule:** Always pair a log action with an explicit allow rule immediately after it. Otherwise, traffic may be denied due to the tier's implicit deny action.  
+    1. **Comprehensive Protection:** Ensure both namespaced and non-namespaced resources are covered by enabling `hostEndpoint` for logging and protecting all traffic in your cluster.  
+
+By strategically placing log actions at the bottom of your policy tiers, you can capture unhandled traffic and identify gaps in your policy configuration. Analyze the logged traffic to gather insights about source, destination, and protocol, and use this data to create targeted policies. Once implemented, validate by checking if the corresponding traffic is no longer logged, indicating the new policies are effective.  
+
+
+### Logging an unprotected cluster 
+
+Kubernetes' default behavior permits all traffic, making early security implementation essential. Calico’s `GlobalNetworkPolicy` with the `Log` action is a powerful tool for monitoring cluster traffic. It allows you to observe, analyze, and secure traffic flows. A single `GlobalNetworkPolicy` can be used to log all traffic across the cluster, providing comprehensive insights for your security strategy.  
+
+
+To log all traffic across the entire cluster, define a GlobalNetworkPolicy:
+```yaml
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+    name: log-everything-in-cluster
+spec:
+    order: 100000
+    tier: default
+    types:
+    - Ingress
+    - Egress
+    egress:
+    - action: Log
+    ingress:
+    - action: Log
+```
+
+### Logging in a namespace
+
+To monitor or troubleshoot traffic within a namespace, use a NetworkPolicy. Apply the policy with the highest priority to capture traffic not handled by existing policies. This provides visibility into unhandled traffic patterns specific to that namespace.
+
+To log all traffic within a specific namespace, create a NetworkPolicy like this:
+```yaml
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: log-everything-in-ns
+  namespace: <YOUR-NAMESPACE>
+spec:
+  order: 100001
+  tier: default
+  types:
+  - Ingress
+  - Egress
+  egress:
+  - action: Log
+  ingress:
+  - action: Log
+```
+
+### Logging all traffic in a cluster
+
+By default, Calico policies apply only to namespaced resources (e.g., Pods, ServiceAccounts). To achieve total protection, enable hostEndpoint support to include non-namespaced resources and host nodes.
```

Comment:
We should caution that BPF log actions are very slow and should only be used for debugging.  As soon as a program logs it seems to add a  huge overhead.

Reply:
Noted.